app/controllers/idv/how_to_verify_controller.rb

```diff
@@ -65,8 +63,8 @@ def self.step_info
           idv_session.service_provider&.in_person_proofing_enabled
         end,
         undo_step: ->(idv_session:, user:) {
-                     idv_session.skip_doc_auth = nil
-                     idv_session.skip_doc_auth_from_how_to_verify = nil
+                     idv_session.skip_doc_auth_from_how_to_verify = nil ||
+                      idv_session.skip_doc_auth = nil
```

Comment:
I'm confused by this and the previous couple of instances. It looks like variable assignments joined with an `OR`? I have a sneaking suspicion that what is happening here is not what is intended.

Reply:
+

Reply:
Yep yep - it is strange. The short "explanation" is that when I originally removed the `idv_session.skip_doc_auth` variable completely we saw a bug during 50/50 state testing (going from this branch to main), where this variable was expected to be set. So we made the awkward but temporary change here, following the recommended `new_var || old_var` pattern, deploy, then remove `old_var` and that seemed to work 🤷 

The longer details can be found in this discussion https://github.com/18F/identity-idp/pull/11043#discussion_r1779126211

In isolation, I am not the most comfortable with this change but given that the very next PR will remove `idv_session.skip_doc_auth` 🤷

Reply:
+1 on 50/50 test. Happy (really; it's the kind of think I enjoy) to pair and help if you'd like.

Reply:
And while I appreciate the value of following patterns, the practice exists to serve us, not the other way around, and in this specific case, I don't think clarity resulted. I'd prefer less puzzling code this time.